### DISCUSSION TITLE
eme_frontend_cancel_booking_action

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#527] [Q&A] [answered]
Hi,
I've coded a function based on the 'eme_frontend_cancel_booking_action' hook. I would like this function to be activated also if an administrator deletes the booking. Is this possible? And which hook should I use?

**Comment by liedekef** [comment]
eme_delete_rsvp_action is the hook you're looking for I guess

**Reply by f14m07** [reply]
According to the doc, this hook is executed after purging a booking from trash. But when the admin deletes a booking or an event the bookings are sent to the trash, so I think this hook is not the good one for me.

**Reply by liedekef** [reply, marked as answer]
True, eme_trash_rsvp_action is the one :-)

**Reply by f14m07** [reply]
This is my snippet
```
add_action('eme_frontend_cancel_booking_action', 'my_eme_carte2_function');
add_action('eme_trash_rsvp_action', 'my_eme_carte2_function');
function my_eme_carte2_function($booking) {
```
When an admin deletes the booking it doesn't work, but when the user deletes his booking the hook is activated twice.
The hook 'eme_trash_rsvp_action' is not documented.

**Reply by liedekef** [reply]
The hook is now documented, but since your code is executed twice it works as expected: once for the eme_frontend_cancel_booking_action and once for eme_trash_rsvp_action
In the backend, the "Delete booking (move to trash)" action calls this action too, so it should get executed fine. Make sure your code is included in the backend too.

**Reply by f14m07** [reply]
Thanks for the doc. The snippet is 'Run snippet everywhere' so the problem is different. I have to look for
